telephony/android_modem.c

```diff
@@ -1935,24 +1977,8 @@ handleRadioPower( const char*  cmd, AModem  modem )
         return "+CME ERROR: 50";
     }
 
-    if (radio_state == modem->radio_state) {
-        return "OK";
-    }
-
-    modem->radio_state = radio_state;
     amodem_reply(modem, "OK");
-
-    switch (radio_state) {
-        case A_RADIO_STATE_OFF:
-            amodem_set_voice_registration(modem, A_REGISTRATION_UNREGISTERED);
-            amodem_set_data_registration(modem, A_REGISTRATION_UNREGISTERED);
-            asimcard_reset_status_after_radio_off(modem->sim);
-            break;
-        case A_RADIO_STATE_ON:
-            amodem_set_voice_registration(modem, A_REGISTRATION_HOME);
-            amodem_set_data_registration(modem, A_REGISTRATION_HOME);
-            break;
-    }
+    _amodem_set_radio_state(modem, radio_state);
```

Comment:
After offline discussion, we have verified that RIL_UNSOL_RESPONSE_RADIO_STATE_CHANGED won't be sent twice, since the unsolicited reponse sending part is moved to another function.